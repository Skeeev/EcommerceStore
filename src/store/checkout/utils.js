import get from 'lodash/get';
import find from 'lodash/find';

const normalizeProducts = products =>
  products.map(({ key, name: title, price, total, quantity, thumb }) => ({
    id: `${key}`,
    title,
    price,
    total,
    quantity,
    imageUri: `http:${thumb}`
  }));

export const extractCheckoutData = ({ products, totals }) => ({
  products: normalizeProducts(products),
  total: get(find(totals, { id: 'total' }), 'text', '$00.00'),
  totalQuantity: products.reduce((total, { quantity }) => total + quantity, 0)
});
