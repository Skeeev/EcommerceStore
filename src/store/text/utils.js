export const prepareTextPayload = text => ({
  payload: {
    text
  }
});
