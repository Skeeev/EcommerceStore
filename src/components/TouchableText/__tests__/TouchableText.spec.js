import React from 'react';
import { shallow } from 'enzyme';

import { getComponentDisplayName } from 'src/utils/displayName';
import { TouchableText } from '..';

describe(`${getComponentDisplayName(TouchableText)} component`, () => {
  const renderComponent = props => shallow(<TouchableText {...props} />);

  it('renders correctly', () => {
    const component = renderComponent({
      children: 'Dummy text'
    });

    expect(component).toMatchSnapshot();
  });
});
