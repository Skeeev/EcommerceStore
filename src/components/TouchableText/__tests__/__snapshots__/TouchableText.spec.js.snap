// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchableText component renders correctly 1`] = `
<AnimatedComponent
  onLongPress={[Function]}
  style={
    Object {
      "transform": Array [
        Object {
          "scale": 1,
        },
      ],
    }
  }
>
  Dummy text
</AnimatedComponent>
`;
